source/fab/build_config.py

```diff
@@ -43,8 +43,15 @@ def __init__(self, project_label: str, source_root: Optional[Path] = None, steps
             by underscores.
         :param source_root:
             Optional argument to allow the config to find source code outside it's project workspace.
-            This is useful, for example, when the :py:mod:`fab.steps.grab <grab>` is in a separate script to be run
-            less frequently. In this scenario, the source code will be found in a different project workspace folder.
+            This is useful, for example, when the :py:mod:`~fab.steps.grab` is in a separate script to be run
+            less frequently. In this scenario, the source code might be found in a different project workspace folder.
+            This argument is intended to be used *instead* of grab steps.
+
+            .. note::
+
+                **Take care if using this argument with grab steps**
+                because you could write to a source folder *outside* your project workspace.
```

Comment:
Good question. Not yet. Some options:
- We could raise an exception if using a grab step when the source folder is overridden. I don't think we've got a use case for using both at the same time; they seem mutually exclusive so that seems safe to do.
- Remove the source folder override. We can just use a folder grab instead and the problem goes away. We added the override during early development when porting FCM build configs. I don't think it's essential.

Reply:
If this problem exists purely because of unneeded functionality then the solution seems clear. Remove the cause of the problem. It may be necessary to document the use of the file system grab in this situation if it is going to be common practice.